paimon-core/src/main/java/org/apache/paimon/iceberg/manifest/IcebergConversions.java

```diff
@@ -74,8 +79,32 @@ public static ByteBuffer toByteBuffer(DataType type, Object value) {
             case DECIMAL:
                 Decimal decimal = (Decimal) value;
                 return ByteBuffer.wrap((decimal.toUnscaledBytes()));
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+                TimestampType timestampType = (TimestampType) type;
+                precision = timestampType.getPrecision();
+                timestamp = (Timestamp) value;
+                return convertTimestampWithPrecisionToBuffer(timestamp, precision);
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                LocalZonedTimestampType localTimestampType = (LocalZonedTimestampType) type;
+                precision = localTimestampType.getPrecision();
+                timestamp = (Timestamp) value;
+                return convertTimestampWithPrecisionToBuffer(timestamp, precision);
```

Comment:
```suggestion
            case TIMESTAMP_WITHOUT_TIME_ZONE:
                TimestampType timestampType = (TimestampType) type;
                return convertTimestampWithPrecisionToBuffer((Timestamp) value, timestampType.getPrecision());
            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
                LocalZonedTimestampType localTimestampType = (LocalZonedTimestampType) type;
                return convertTimestampWithPrecisionToBuffer((Timestamp) value, localTimestampType.getPrecision());
```

So that local variables at the beginning can be removed.

Reply:
Thank you. I have learned to be rigorous, meticulous and detail-oriented